src/librbd/ImageCtx.h

```diff
@@ -68,6 +68,7 @@ namespace librbd {
                                         // a format librados can understand
     std::map<librados::snap_t, SnapInfo> snap_info;
     std::map<std::string, librados::snap_t> snap_ids;
+    std::map<cls::rbd::SnapshotNamespace, librados::snap_t> snap_namespace_ids;
```

Comment:
@dillaman I created another PR #12970 for this refactoring. I would like to move through that PR gradually. Could you please take a look at the current version. It has only changes to ImageCtx. Please take a look and tell me if this is what you want to do to ImageCtx in principle. Thanks.